### ISSUE TITLE
Tensorflow support for Nvidia geforce rtx 3090 - CUDA-11.1

### ISSUE THREAD

**Issue by JVedant** [tensorflow/tensorflow#44682]
The rtx 3090 has been a beast in deeplearning performance and yet tensorflow has no support for training on CUDA-11.1 and cuDNN-8.

it would be great if added soon.

**Comment by Saduf2019** [comment]
@JVedant 
We see that the issue template has not been filled, could you please do so as it helps us analyse the issue [tf version, steps followed before you ran into this error or stand alone code to reproduce the issue faced]
This issue is already reported, please refer to these existing issues with respect to rtx 3090 and CUDA 11.1  #44200, #44251, #43588

**Comment by JVedant** [comment]
It's my first time opening an issue here so i'm not aware about the issue template.

I think i've got some help by checking the issue id's provided by you.
I'll try it once and if I wont be able to achieve any results, i'll open an issue with 'build' tag.

Thanks for providing the solution!

**Comment by Saduf2019** [comment]
@JVedant 
Thank you for the update, please move the ticket to closed status if resolved.

**Comment by fshofmann** [comment]
@JVedant I got my 3090 yesterday and was in full panic mode as well, but if you have not installed everything here is my order that "worked" (atleast my custom training loops and non sequential models produce results that I expect, but with a major speedup compared to my old setup):

https://github.com/tensorflow/tensorflow/issues/44159

The OP is a bit older, so the versions have slightly changed:

- OS Platform and Distribution: Windows 10 Pro 64 bit
- TensorFlow installed from (source or binary): pip install tf-nightly-gpu=2.5.0.dev20201110
- Python version: 3.8.6 (don't forget to downgrade numpy because windows reasons)
- Installed using virtualenv? pip? conda?: Pip (in conda env)
- CUDA/cuDNN version: CUDA 11.1/cuDNN 8.4.0.30 *****
- GPU model and memory: NVIDIA RTX 3090 (24 GB)


***** As with the issue I simply installed CUDA 10.2 for the missing DLL file. Simple install, no change in env CUDA paths (these are changed automatically but do no matter?), no additional CUDNN, nothing. Afterwards it worked for me. Some in the other issue simply downloaded the missing DLL (10 versio nof cuda solver) and put it into their 11 folder to get it working.

**Comment by Thunder003** [comment]
@fshofmann is your T.F. 2.5.0.dev20201110 recognized the 3090 GPU?

I tried running:-

from tensorflow.python.client import device_lib
device_lib.list_local_devices()

but the output was Just CPU.

**Comment by fshofmann** [comment]
@Thunder003 My TF installation definitely uses the GPU. Are you 100% sure that you installed the package tf-nightly-gpu=2.5.0.dev20201110 and not the gpu-less version tf-nightly?

**Comment by Thunder003** [comment]
@fshofmann yes, I used the command: **pip install tf-nightly-gpu==2.5.0.dev20201028**,......then check whether GPU recognized by TF or not using:-

**from tensorflow.python.client import device_lib
device_lib.list_local_devices()**

which returned me CPU only. Did you use the above line of code to confirm whether the T.F. recognized the GPU? if not, then can you give a shot to those two lines of code and see whether it returns GPU or not,  maybe that code no longer works on new T.F. version which is why it is not returning any GPU. In such a case, do you know any command to check whether T.F. is recognizing GPU or not.

**Comment by JVedant** [comment]
> @fshofmann yes, I used the command: **pip install tf-nightly-gpu==2.5.0.dev20201028**,......then check whether GPU recognized by TF or not using:-
> 
> **from tensorflow.python.client import device_lib device_lib.list_local_devices()**
> 
> which returned me CPU only. Did you use the above line of code to confirm whether the T.F. recognized the GPU? if not, then can you give a shot to those two lines of code and see whether it returns GPU or not, maybe that code no longer works on new T.F. version which is why it is not returning any GPU. In such a case, do you know any command to check whether T.F. is recognizing GPU or not.

You can give a try to tf.test.is_gpu_available() "i know it's gonna get depricated in new versions but it works as of now".

**Comment by fshofmann** [comment]
> @fshofmann yes, I used the command: **pip install tf-nightly-gpu==2.5.0.dev20201028**,......then check whether GPU recognized by TF or not using:-
> 
> **from tensorflow.python.client import device_lib device_lib.list_local_devices()**
> 
> which returned me CPU only. Did you use the above line of code to confirm whether the T.F. recognized the GPU? if not, then can you give a shot to those two lines of code and see whether it returns GPU or not, maybe that code no longer works on new T.F. version which is why it is not returning any GPU. In such a case, do you know any command to check whether T.F. is recognizing GPU or not.

I am usually using the following snippet at the start of my programs to double check my env:
`if tf.test.gpu_device_name():
    print('Default GPU Device: {}'.format(tf.test.gpu_device_name()))
else:
    print("Please install GPU version of TF")`

In addition, for the past 12 days, I have been fully utilizing the new GPU to do GAN training, so it definetely works (it is constantly at 100% load) :) That's why I am so sure it works. 

As I said before, this issue helped me a lot: https://github.com/tensorflow/tensorflow/issues/44159
It practically described what I already said here.

**Comment by Thunder003** [comment]
@JVedant @fshofmann thanks for your answer. I figured out that it is libcusolver.so.10 issue( It is not present), I followed #43947 of creating a symlink https://github.com/tensorflow/tensorflow/issues/43947#issuecomment-715295153 but I'm getting another error:-
""failed call to cuInit: CUDA_ERROR_SYSTEM_DRIVER_MISMATCH: system has unsupported display driver / cuda driver combination"". Do you know the solution for it or else I'll ask on #43947

**Comment by JVedant** [comment]
> @JVedant @fshofmann thanks for your answer. I figured out that it is libcusolver.so.10 issue( It is not present), I followed #43947 of creating a symlink [#43947 (comment)](https://github.com/tensorflow/tensorflow/issues/43947#issuecomment-715295153) but I'm getting another error:-
> ""failed call to cuInit: CUDA_ERROR_SYSTEM_DRIVER_MISMATCH: system has unsupported display driver / cuda driver combination"". Do you know the solution for it or else I'll ask on #43947

I guess you've got cuda 11.1, if yes then find libcusolver.so.11 in your cuda path and make a copy of it named 'libcusolver.so.10' and paste it in your environment at **env -> lib -> python3.x -> site-packages -> tensorflow -> python** it works.

For me CUDA 11.1, cuDNN-8 and tf-2.4.0-rc1 worked perfectly.

**Comment by Thunder003** [comment]
@JVedant thanks for your answer. My bug is fixed using by taking some references from https://github.com/NVIDIA/nvidia-docker/issues/1256

**Comment by linuxmaster0312** [comment]
> > @fshofmann yes, I used the command: **pip install tf-nightly-gpu==2.5.0.dev20201028**,......then check whether GPU recognized by TF or not using:-
> > **from tensorflow.python.client import device_lib device_lib.list_local_devices()**
> > which returned me CPU only. Did you use the above line of code to confirm whether the T.F. recognized the GPU? if not, then can you give a shot to those two lines of code and see whether it returns GPU or not, maybe that code no longer works on new T.F. version which is why it is not returning any GPU. In such a case, do you know any command to check whether T.F. is recognizing GPU or not.
> 
> You can give a try to tf.test.is_gpu_available() "i know it's gonna get depricated in new versions but it works as of now".
> 
> you can check #44753 where I've mentioned the system config and #43947 is where i got my solution. If you're still facing the issue lemme know on here. I can write about the whole process i followed to get it running.

@JVedant could you give a more detailed description on how you got tensorflow to train on your gpu? I have tried tf-nightly-gpu and the 2.4.0 release candidates but I still cannot get my models to train on my gpu. Details at my [stack question](https://stackoverflow.com/questions/65056018/tensorflow-trains-on-cpu-instead-of-rtx-3000-series-gpu?noredirect=1#comment115013539_65056018)

**Comment by JVedant** [comment]
> > > @fshofmann yes, I used the command: **pip install tf-nightly-gpu==2.5.0.dev20201028**,......then check whether GPU recognized by TF or not using:-
> > > **from tensorflow.python.client import device_lib device_lib.list_local_devices()**
> > > which returned me CPU only. Did you use the above line of code to confirm whether the T.F. recognized the GPU? if not, then can you give a shot to those two lines of code and see whether it returns GPU or not, maybe that code no longer works on new T.F. version which is why it is not returning any GPU. In such a case, do you know any command to check whether T.F. is recognizing GPU or not.
> > 
> > 
> > You can give a try to tf.test.is_gpu_available() "i know it's gonna get depricated in new versions but it works as of now".
> > you can check #44753 where I've mentioned the system config and #43947 is where i got my solution. If you're still facing the issue lemme know on here. I can write about the whole process i followed to get it running.
> 
> @JVedant could you give a more detailed description on how you got tensorflow to train on your gpu? I have tried tf-nightly-gpu and the 2.4.0 release candidates but I still cannot get my models to train on my gpu. Details at my [stack question](https://stackoverflow.com/questions/65056018/tensorflow-trains-on-cpu-instead-of-rtx-3000-series-gpu)

can you please mention your OS. you can also try **tf.test.is_gpu_available()** to check whether the gpu is detected by tf or not, also try **tf.device('/gpu:0')** to force the training on gpu, there is a chance of the model not performing backward propagation (accuracy and loss won't change even if the model is getting trained). if you observe the accuracy not changing, i would recommend to reinstall everything including nvidia drivers, cudnn, cuda and tf(tf-nightly with gpu).

If you face any issues as mentioned in  #43947 , you can perform steps mentioned in [above](https://github.com/tensorflow/tensorflow/issues/44682#issuecomment-732283634) solutions

(Also maybe you'll have to perform some manual changes in the folder directory if nothing works) if any such things happen you can reply to this ticket and we can try something else.

**Comment by linuxmaster0312** [comment]
Thanks for the tips, I am using windows 10 home.
